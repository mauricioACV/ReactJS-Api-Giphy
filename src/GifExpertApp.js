import { useState } from "react";
import {AddCategory} from './Components/AddCategory';
import { GifGrid } from "./Components/GifGrid";

export const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['Mauricio', 'Javiera', 'Alejandra'];

    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     //Una forma de hacerlo, una de las mÃ¡s comunes
    //     // setCategories([...categories, 'Constanza']);

    //     //otra forma de hacerlo, usando el callback de setCategories
    //     // setCategories( cats => [...cats, 'Conztanza']);
    // }

    return ( 
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories}/>
        <hr></hr>

        {/* <button onClick={ handleAdd }>Agregar</button> */}

        <ol>
            {
                categories.map((category) => (
                    <GifGrid 
                        key={category}
                        category={category}
                    />
                ))
            }
        </ol>
        </>
     );
}