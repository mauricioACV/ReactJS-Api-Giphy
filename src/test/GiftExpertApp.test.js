import React from 'react'
import {shallow} from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en componente <GifExpertApp />', () => {
    
    test('debe de mostrar el componente correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['one punch', 'dragon ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
    
})