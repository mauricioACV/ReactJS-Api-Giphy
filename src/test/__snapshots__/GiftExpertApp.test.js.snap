// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifExpertApp /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;
